inspect_builder.go

```diff
@@ -48,15 +54,50 @@ func (c *Client) InspectBuilder(name string, daemon bool) (*BuilderInfo, error)
 		}
 	}
 
+	var bpLayers dist.BuildpackLayers
+	if _, err := dist.GetLabel(img, dist.BuildpackLayersLabel, &bpLayers); err != nil {
+		return nil, err
+	}
+
 	return &BuilderInfo{
 		Description:     bldr.Description(),
 		Stack:           bldr.StackID,
 		Mixins:          append(commonMixins, buildMixins...),
 		RunImage:        bldr.Stack().RunImage.Image,
 		RunImageMirrors: bldr.Stack().RunImage.Mirrors,
-		Buildpacks:      bldr.Buildpacks(),
+		Buildpacks:      uniqueBuildpacks(bldr.Buildpacks()),
 		Order:           bldr.Order(),
+		BuildpackLayers: bpLayers,
 		Lifecycle:       bldr.LifecycleDescriptor(),
 		CreatedBy:       bldr.CreatedBy(),
 	}, nil
 }
+
+func uniqueBuildpacks(buildpacks []dist.BuildpackInfo) []dist.BuildpackInfo {
+	buildpacksSet := map[BuildpackInfoKey]int{}
+	homePageSet := map[BuildpackInfoKey]string{}
+	for _, buildpack := range buildpacks {
+		key := BuildpackInfoKey{
+			ID:      buildpack.ID,
+			Version: buildpack.Version,
+		}
+		_, ok := buildpacksSet[key]
+		switch {
+		case ok && buildpack.Homepage != "":
+			homePageSet[key] = buildpack.Homepage
```

Comment:
Good catch, think this can probably go its actually not tested. Original thinking was we have a repeated buildpack, in the builder label. One entry has a `homepage`, in the other it is omitted. We should preserve the `homepage` from on the buildpack where it is present.

Thinking about this, it's probably impossible for a single combination of buildpack `id` and `version` to have a different `homepage` metadata in the first place.

Reply:
I'm not sure if it's impossible, but if they do have different homepages, that's probably a mistake, so I'm not worried about preserving any particular one of them.